src/platform_impl/windows/event_loop/runner.rs

```diff
@@ -1,51 +1,50 @@
-use std::{any::Any, cell::RefCell, collections::VecDeque, mem, panic, ptr, rc::Rc, time::Instant};
+use std::{
+    any::Any,
+    cell::{Cell, RefCell},
+    collections::{HashSet, VecDeque},
+    mem, panic, ptr,
+    rc::Rc,
+    time::Instant,
+};
 
-use winapi::{shared::windef::HWND, um::winuser};
+use winapi::{
+    shared::{minwindef::DWORD, windef::HWND},
+    um::winuser,
+};
 
 use crate::{
     dpi::PhysicalSize,
     event::{Event, StartCause, WindowEvent},
     event_loop::ControlFlow,
-    platform_impl::platform::{event_loop::EventLoop, util},
+    platform_impl::platform::util,
     window::WindowId,
 };
 
 pub(crate) type EventLoopRunnerShared<T> = Rc<ELRShared<T>>;
 pub(crate) struct ELRShared<T: 'static> {
-    runner: RefCell<Option<EventLoopRunner<T>>>,
-    buffer: RefCell<VecDeque<BufferedEvent<T>>>,
-    redraw_buffer: Rc<RefCell<VecDeque<WindowId>>>,
-}
-struct EventLoopRunner<T: 'static> {
-    control_flow: ControlFlow,
-    runner_state: RunnerState,
-    modal_redraw_window: HWND,
-    in_modal_loop: bool,
-    event_handler: Box<dyn FnMut(Event<'_, T>, &mut ControlFlow)>,
-    panic_error: Option<PanicError>,
-    redraw_buffer: Rc<RefCell<VecDeque<WindowId>>>,
+    thread_msg_target: HWND,
+    wait_thread_id: DWORD,
+    processing_events: Cell<ProcessingEvents>,
+    panic_error: Cell<Option<PanicError>>,
+    control_flow: Cell<ControlFlow>,
+    last_events_cleared: Cell<Instant>,
+    event_handler: Cell<Option<Box<dyn FnMut(Event<'_, T>, &mut ControlFlow)>>>,
+    event_buffer: RefCell<VecDeque<BufferedEvent<T>>>,
+    owned_windows: Cell<HashSet<HWND>>,
```

Comment:
EventLoopRunner has quite a bit more state than before. Compare it to my PR:
```
struct EventLoopRunner<T: 'static> {
    control_flow: ControlFlow,
    runner_state: RunnerState,
    modal_redraw_window: HWND,
    in_modal_loop: bool,
    event_handler: Box<dyn FnMut(Event<'_, T>, &mut ControlFlow)>,
    panic_error: Option<PanicError>,
}
```

Reply:
I don't think that's a fair comparison. This PR introduces a few more fields, but the actual state management code has been simplified considerably; compare the [old state code](https://github.com/filnet/winit/blob/f346b43bd2f1f5292fc9d42ac5154afce2c9c1ae/src/platform_impl/windows/event_loop/runner.rs#L261-L455), which contains *several* nested match statements and is spread out over multiple functions, with the [new state code](https://github.com/filnet/winit/blob/f346b43bd2f1f5292fc9d42ac5154afce2c9c1ae/src/platform_impl/windows/event_loop/runner.rs#L261-L455), which concentrates all the state transition code into a single, more easily understandable function.

Reply:
On the other hand you removed EventLoopRunner which had some state too (?).

Reply:
I've flattened out `EventLoopRunner` and `ELRShared` into a single struct. Thinking about it, we should probably the new `ELRShared` to `EventLoopRunner`, since the only real reason the `ELRShared` name exists was to distinguish between it and the normal `EventLoopRunner`.

Reply:
Renaming `ELRShared` to `EventLoopRunner` sounds good.